parsigs/parse_sig_api.py

```diff
@@ -86,27 +87,39 @@ def _parse_sig(sig: str, model: Language):
 
     return _create_structured_sigs(model_output)
 
+def _autocorrect(sig):
+    spell = SpellChecker()
+    spell.word_frequency.load_text_file(os.path.join('data', 'drug_names.txt'))
+    corrected_words = [spell.correction(word) if not any(char.isdigit() for char in word) else word for word in sig.split()]
+    sig = ' '.join(corrected_words)
+    return sig
 
-def _pre_process(sig):
-    sig = sig.lower().replace('twice', '2 times').replace("once", '1 time').replace("nightly", "every night")
-
-    sig = _add_space_around_parentheses(sig)
-
-    # remove extra spaces between words
-    sig = re.sub(r'\s+', ' ', sig)
-
-    output_words = []
-    words = sig.split()
-    for word in words:
-        if word == 'tab':
-            word = word.replace('tab', 'tablet')
-            output_words.append(word)
-        else:
-            output_words.append(word)
-    sig = ' '.join(output_words)
 
+def _pre_process(sig):
+    #text to lower case and remove leading and ending whitespace
+    sig = sig.lower().strip()
+    # Spell check
+    sig = _autocorrect(sig)
+    #manual replacements of familiar phrases and characters
+    replacements = {
```

Comment:
This does not contain the previoud logic with the regex, also why there are spaces before and after the words? 
I think it is better using the old logic, or recreating it in a dictionary similar to what you did but globally as a constant

Reply:
I've changed the replacements dictionary to preserve the same desired outcome.